### PR TITLE
Moved setPrimaryService into Accessory Constructor

### PR DESCRIPTION
**Pull request by luc-ass**
This fixes the vacuum not being displayed as a fan as reported here #542.

@afharo, please have a look at this. I am not aware of negative side effects, but there may be some I didn't catch.

### CODE REVIEW

**Comment by luc-ass** [comment]
Checks resulted in an Error: `TypeError: this.service.setPrimaryService is not a function`

**Comment by luc-ass** [comment]
Something is still off. Seems like the services are resorted like crazy. Any ideas?

https://user-images.githubusercontent.com/3263257/193253056-e977121e-5113-4ce5-8b98-72f2abc84e69.MOV

File: src/services/main_service.ts [hunk 1]
```diff
@@ -28,12 +28,12 @@ export class MainService extends PluginServiceClass {
         .getCharacteristic(this.hap.Characteristic.RotationSpeed)
         .onGet(() => this.getSpeed())
         .onSet((newState) => this.setSpeed(newState));
+      this.service.setPrimaryService;
     } else {
       this.service = new this.hap.Service.Switch(this.config.name, "Vacuum");
+      this.service.setPrimaryService;
     }
 
-    if (this.service.setPrimaryService) this.service.setPrimaryService(true);
```
Comment (afharo) [inline src/services/main_service.ts:35]: In some versions of homebridge, `setPrimaryService` is not available, that's why I had to add that validation in the past.

If the method exists, call it.

File: src/services/main_service.ts [hunk 2]
```diff
@@ -28,12 +28,12 @@ export class MainService extends PluginServiceClass {
         .getCharacteristic(this.hap.Characteristic.RotationSpeed)
         .onGet(() => this.getSpeed())
         .onSet((newState) => this.setSpeed(newState));
+      this.service.setPrimaryService;
     } else {
       this.service = new this.hap.Service.Switch(this.config.name, "Vacuum");
+      this.service.setPrimaryService;
```
Comment (afharo) [inline src/services/main_service.ts:34]: AFAIK, this is a function, and this line is not calling it. It should be called as `this.service.setPrimaryService(true)`

**Review by afharo — COMMENTED** [review]
IMO, I don't think these changes will fix the issue. 

As you commented earlier, Home will still present them randomly.

I think this can be fixed in the getServices method instead.

I may be able to spend a couple of hours on it this weekend and have a proposal.

File: src/services/main_service.ts (hunk 2 above)
Comment (luc-ass) [reply src/services/main_service.ts:34]: That's what I tried at first, but the checks did fail so it wasn't able to merge that way.

**Comment by luc-ass** [comment]
resolved as of #572

**Pull request closed without merging**